cmd/controller-manager/app/options/options.go

```diff
@@ -106,6 +111,7 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.InheritOwnerAnnotations, "inherit-owner-annotations", true, "Enable inherit owner annotations for pods when create podgroup; it is enabled by default")
 	fs.Uint32Var(&s.WorkerThreadsForPG, "worker-threads-for-podgroup", defaultPodGroupWorkers, "The number of threads syncing podgroup operations. The larger the number, the faster the podgroup processing, but requires more CPU load.")
 	fs.Uint32Var(&s.WorkerThreadsForGC, "worker-threads-for-gc", defaultGCWorkers, "The number of threads for recycling jobs. The larger the number, the faster the job recycling, but requires more CPU load.")
+	fs.StringVar(&s.ControllerGates, "controller-gates", "*", "Specify controller gates. Use '*' for all controllers, '+queue-controller,pg-controller' to start specific controllers, and '-jobflow-controller,jobtemplate-controller' to stop specific controllers.")
```

Comment:
`StringSliceVar` is better.

Reply:
And we should all print all controllers here to let uer know what controllers we have.

Reply:
done